Create motor_controller_node

matthewia94 opened the pull request:
The motor_controller_node will convert twist messages to motor
velocities and send them to the boards which are controlling the motors.
They are currently arbitrarily scaled and will need to be adjusted later
based on the robots speed.

Closes #5

matthewia94 committed:
Create motor_controller_node

The motor_controller_node will convert twist messages to motor
velocities and send them to the boards which are controlling the motors.
They are currently arbitrarily scaled and will need to be adjusted later
based on the robots speed.

Closes #5

islamelnabarawy commented on ros_ws/src/control/src/control/motor_controller.cpp:
```diff
@@ -0,0 +1,26 @@
+void MotorController::twistCallback(const geometry_msgs::Twist::ConstPtr& msg)
+{
+    right_vel.data = (2 * msg->linear.x + msg->angular.z * ROBOT_BASE) / (2 * WHEEL_RAD);
+    left_vel.data = (2 * msg->linear.x + msg->angular.z * ROBOT_BASE) / (2 * WHEEL_RAD);
```
There's an error in the math here. Please double-check it.

Also please refactor the math equations into separate functions so it would be clear what the calculation represents.

islamelnabarawy commented on ros_ws/src/common/constants.h:
```diff
@@ -59,6 +67,10 @@ const std::string TELEOP_NODE = "teleop_control";
+/**
+ * @brief The node name for the motor_controller node
+ */
+const std::string MOTOR_CONTROLLER_NODE = "motor_controller_node";
```
It probably makes sense for us to separate the node names into their own file and put them in a namespace. This doesn't have to happen now for this pull request, but it should happen sometime in the near future.

matthewia94 replied:
Now that I think about this a little bit it might actually make sense to have the node names be literals.  The way we have it right now makes the entire workspace necessary to run a package.  I think might be a good idea to follow the ROS idea that packages should be atomic and interchangeable regardless of workspace.

islamelnabarawy replied:
That's an interesting point!

I think a good middle ground might be to have a separate constants.h file for each package. We will have to duplicate the topic name constants and node names and such, but each node will be an independent unit.

Thoughts?

@ryanloeffelman and @Cazadorro: What do you guys think?

islamelnabarawy commented on ros_ws/src/common/constants.h:
```diff
@@ -66,4 +78,15 @@ const std::string TELEOP_MUX_NODE = "teleop_mux_node";
+/**
+ * @brief The radius of the robot's wheels
+ */
+const double WHEEL_RAD = 0.19;
```
These values should be moved to parameter configurations in the parameter server. We can keep it as-is for now, but please create an issue for it so we can do it in the near future.

islamelnabarawy replied:
Note: Issue #25 created in response to this comment.

matthewia94 committed:
Fix velocity equations

The velocity equations were the same for both wheels previously and
they have been fixed to reflect the differential drive model.  The
equations have also been seperated into functions to make the use of
the equations more clear.
